Source/Core/Core/HW/WiimoteReal/IOdarwin.mm

```diff
@@ -45,6 +46,32 @@ @interface ConnectBT: NSObject {}
 	IOPMAssertionID m_pm_assertion;
 };
 
+class WiimoteDarwinHid final : public Wiimote
+{
+public:
+	WiimoteDarwinHid(IOHIDDeviceRef device);
+	~WiimoteDarwinHid() override;
+
+protected:
+	bool ConnectInternal() override;
+	void DisconnectInternal() override;
+	bool IsConnected() const override;
+	void IOWakeup() override;
+	void IOCancel() override;
+	int IORead(u8* buf) override;
+	int IOWrite(u8 const* buf, size_t len) override;
+
+private:
+	static void ReportCallback(void* context, IOReturn result, void* sender, IOHIDReportType type, uint32_t reportID, uint8_t* report, CFIndex reportLength);
```

Comment:
I'm using uint32_t and uint8_t due to the callback API--but they're probably the same type so I can change it.